### ISSUE TITLE
Issue with ShowAssetPreview and SerializeField

### ISSUE THREAD

**Issue by darktalestudio** [dbrizov/NaughtyAttributes#13]
Hey there,

I've encountered an issue when using ShowAssetPreview together with SerializeField on a Sprite within a scriptable object.

Here is the line:
[SerializeField] [BoxGroup(GENERAL_SETTINGS_HEADER)] [ShowAssetPreview] Sprite itemIcon;

It works when I remove the SerializeField option and use a public modifier instead:
[BoxGroup(GENERAL_SETTINGS_HEADER)] [ShowAssetPreview] public Sprite itemIcon;

here is the Stacktrace:

NullReferenceException: Object reference not set to an instance of an object
NaughtyAttributes.Editor.PropertyUtility.GetAttributes[ShowAssetPreviewAttribute] (UnityEditor.SerializedProperty property) (at Assets/Plugins/NaughtyAttributes/Scripts/Editor/Utility/PropertyUtility.cs:20)
NaughtyAttributes.Editor.ShowAssetPreviewPropertyDrawer.DrawProperty (UnityEditor.SerializedProperty property) (at Assets/Plugins/NaughtyAttributes/Scripts/Editor/PropertyDrawers/ShowAssetPreviewPropertyDrawer.cs:20)
NaughtyAttributes.Editor.InspectorEditor.DrawField (System.Reflection.FieldInfo field) (at Assets/Plugins/NaughtyAttributes/Scripts/Editor/Editors/InspectorEditor.cs:232)
NaughtyAttributes.Editor.InspectorEditor.ValidateAndDrawField (System.Reflection.FieldInfo field) (at Assets/Plugins/NaughtyAttributes/Scripts/Editor/Editors/InspectorEditor.cs:190)
NaughtyAttributes.Editor.InspectorEditor.ValidateAndDrawFields (IEnumerable`1 fields) (at Assets/Plugins/NaughtyAttributes/Scripts/Editor/Editors/InspectorEditor.cs:182)
NaughtyAttributes.Editor.InspectorEditor.OnInspectorGUI () (at Assets/Plugins/NaughtyAttributes/Scripts/Editor/Editors/InspectorEditor.cs:124)
UnityEditor.InspectorWindow.DrawEditor (UnityEditor.Editor[] editors, Int32 editorIndex, Boolean rebuildOptimizedGUIBlock, System.Boolean& showImportedObjectBarNext, UnityEngine.Rect& importedObjectBarRect) (at C:/buildslave/unity/build/Editor/Mono/Inspector/InspectorWindow.cs:1245)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

**Comment by dbrizov** [comment]
Thanks mate, I will fix it

**Comment by darktalestudio** [comment]
Thanks a ton! Have a great day.

**Comment by dbrizov** [comment]
Mate, I am trying to reproduce it. I created a scriptable object and copy-pasted your line, but I don't have an exception. Is the scriptable object nested/linked somehow?

**Comment by darktalestudio** [comment]
Sorry for the late reply, I have been out of town the last few days. The line is part of an abstract class which is inherited by other classes. Take a look at the hierarchy:

public abstract class BaseItem : ScriptableObject
{
[...]
[SerializeField] [BoxGroup(GENERAL_SETTINGS_HEADER)] Sprite icon;
[...]
}

public abstract class EquippableItem : BaseItem
{
[...]
}

public class WeaponConfiguration : EquippableItem
{
[...]
}

So long story short, the WeaponConfiguration is the Scriptable object that is finally shown in the inspector and this is when the exception is thrown.

**Comment by dbrizov** [comment]
Ok, thank :)

**Comment by darktalestudio** [comment]
Any news mate?

**Comment by dbrizov** [comment]
I haven't checked it mate, sorry. I was on a little vacation. Will fix it today/tomorrow

**Comment by dbrizov** [comment]
Fixed it mate. When I was trying to get the field info via reflection the "private" fields weren't accessible, because child classes can't see them. Now I am getting only the declared fields of each class. I am doing this for all classes in the hierarchy while there is a parent class.

**Comment by darktalestudio** [comment]
Confirmed working! Good job mate.



Von Samsung-Tablet gesendet

-------- Ursprüngliche Nachricht --------
Von: Denis Rizov <notifications@github.com>
Datum: 05.03.18 16:21 (GMT+01:00)
An: dbrizov/NaughtyAttributes <NaughtyAttributes@noreply.github.com>
Cc: darktalestudio <norman.krueger@live.nl>, Author <author@noreply.github.com>
Betreff: Re: [dbrizov/NaughtyAttributes] Issue with ShowAssetPreview and SerializeField (#13)


Fixed it mate. When I was trying to get the field info via reflection the "private" fields weren't accessible, because child classes can't see them. Now I am getting only the declared fields of each class. I am doing this for all classes in the hierarchy while there is a parent class.
